runtime/src/test/java/com/dylibso/chicory/testing/ChicoryTestWatcher.java

```diff
@@ -24,6 +24,9 @@ public void testDisabled(ExtensionContext context, Optional<String> optional) {
 
     @Override
     public void testFailed(ExtensionContext context, Throwable throwable) {
+        if (context.getDisplayName().endsWith("Instance")) {
+            // TODO: stop executing the rest of the test class
```

Comment:
@thomasdarimont if you want to pick this one up it will be great.
The motivation is that if an `Instance` creation fails, the following tests need to be skipped, otherwise they will throw NPEs.
It's a quality of life for developers, but it will be appreciated if it's a low-hanging fruit.

Reply:
Thanks for the hint, I'll give it a spin.

Reply:
@andreaTP I found a JUnit5 native way to do it, shall I just add the commit to this branch?

Reply:
```java
package com.dylibso.chicory.testing;

import java.lang.reflect.Method;
import java.util.Base64;
import java.util.Optional;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.ConcurrentMap;
import org.junit.jupiter.api.Tag;
import org.junit.jupiter.api.extension.ExtensionContext;
import org.junit.jupiter.api.extension.InvocationInterceptor;
import org.junit.jupiter.api.extension.ReflectiveInvocationContext;
import org.junit.jupiter.api.extension.TestWatcher;

/**
 * JUnit {@link TestWatcher} extension for analyzing test outcomes of Web Assembly module tests.
 */
public class ChicoryTestWatcher implements TestWatcher, InvocationInterceptor {

    private static final ConcurrentMap<String, String> TEST_CLASSES_TO_SKIP = new ConcurrentHashMap<>();

    public void interceptTestMethod(
            Invocation<Void> invocation,
            ReflectiveInvocationContext<Method> invocationContext,
            ExtensionContext extensionContext)
            throws Throwable {

        String testFqcn = extensionContext.getTestClass().get().getName();
        if (TEST_CLASSES_TO_SKIP.containsKey(testFqcn)) {
            System.err.println("Skipping remaining test methods for " + testFqcn);
            invocation.skip();
        } else {
            invocation.proceed();
        }
    }

    @Override
    public void testAborted(ExtensionContext context, Throwable throwable) {
        // do something
    }

    @Override
    public void testDisabled(ExtensionContext context, Optional<String> optional) {
        // do something
    }

    @Override
    public void testFailed(ExtensionContext context, Throwable throwable) {
        if (context.getDisplayName().endsWith("Instance()")) {
            skipExecutingRemainingTestMethods(context);
        }
        createWasmObjectDumpForFailedTest(context);
    }

    private static void skipExecutingRemainingTestMethods(ExtensionContext context) {
        String testFqcn = context.getTestClass().get().getName();
        System.err.printf(
                "Disable remaining tests for %s (%s)", context.getDisplayName(), testFqcn);
        TEST_CLASSES_TO_SKIP.put(testFqcn, testFqcn);
    }

    private void createWasmObjectDumpForFailedTest(ExtensionContext context) {

        Method testMethod = context.getRequiredTestMethod();
        if (!isWasmObjectDumpEnabledFor(testMethod)) {
            return;
        }

        Class<?> clazz = context.getRequiredTestClass();
        String reference = clazz.getSimpleName() + "." + testMethod.getName();
        System.err.printf("wasm-objdump for test: %s%n", reference);

        String wasmFile = null;
        String symbolFilter = null;
        Tag[] tagsAnno = testMethod.getAnnotationsByType(Tag.class);
        for (var tag : tagsAnno) {
            String tagValue = tag.value();
            if (tagValue.startsWith("export=")) {
                symbolFilter =
                        new String(
                                Base64.getDecoder().decode(tagValue.substring("export=".length())));
            }
            if (tagValue.startsWith("wasm=")) {
                wasmFile = tagValue.substring("wasm=".length());
            }
        }

        WasmObjDumpTool.dump(wasmFile, symbolFilter);
    }

    private boolean isWasmObjectDumpEnabledFor(Method testMethod) {

        var methodAnnotation = testMethod.getAnnotation(ChicoryTest.class);
        if (methodAnnotation != null) {
            return methodAnnotation.dumpOnFail();
        }

        var typeAnnotation = testMethod.getDeclaringClass().getAnnotation(ChicoryTest.class);
        if (typeAnnotation != null) {
            return typeAnnotation.dumpOnFail();
        }

        return false;
    }

    @Override
    public void testSuccessful(ExtensionContext extensionContext) {
        // do something
    }
}
```

Tested this via:
```java
package com.dylibso.chicory.demo;

import com.dylibso.chicory.testing.ChicoryTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Assumptions;
import org.junit.jupiter.api.MethodOrderer;
import org.junit.jupiter.api.Order;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.TestInstance;
import org.junit.jupiter.api.TestMethodOrder;

@ChicoryTest
@TestMethodOrder(MethodOrderer.OrderAnnotation.class)
@TestInstance(TestInstance.Lifecycle.PER_CLASS)
public class SkipTestsExample {

    @Test
    @Order(1)
    void testFoo1Instance() {
        Assertions.fail();
    }

    @Test
    @Order(2)
    void testFoo2Instance() {
        // should not be executed
        Assertions.fail();
    }

    @Test
    @Order(3)
    void testFoo3Instance() {
        // should not be executed
        Assertions.fail();
    }
}
```

With the additional `InvocationInterceptor` we only the first test is executed, but the remaining tests of the test-class are skipped.

I'm not sure about the Filter criterion, perhaps we could use a `Tag` on class level to toggle this behavior, e.g.: `@Tag("onFailure=skipRemainingTests)`.

Reply:
Uhm, maybe my suggestion was a bad idea.
Thanks for checking, but probably it's better to avoid this complexity only for debugging purposes.
So sorry 😔

Reply:
No worries :) was a fun experiment.

Perhaps there is a better way to do it... I'll keep an eye on it. If you have other suggestions for low-hanging fruits let me know :)